src/linked_list.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+"""This file implements a basic linked list data structure."""
+from __future__ import unicode_literals
+
+
+class Node(object):
+    """This builds our Node class."""
+
+    def __init__(self, data):
+        """Initial setup of Node objects."""
+        self.data = data
+        self.next_node = None
+
+
+class LinkedList(object):
+    """This builds LinkedList class."""
+
+    def __init__(self, data=None):
+        """Initial setup of LinkedList class."""
+        self.head = None
+        try:
+            if data:
+                for item in data:
+                    self.push(item)
+        except TypeError:
+            raise TypeError('Please enter an object that is iterable.')
+
+    def push(self, data):
+        """Function builds push method of LinkedList class."""
```

Comment:
A good docstring should tell us what the item it documents *does*.  This describes the effect of evaluating the function definition.  Something like `"""Add a new node containing `data` to the head of the list"""`